src/inmanta/agent/agent.py

```diff
@@ -886,39 +893,53 @@ async def prepare_resources(
         """
         started = datetime.datetime.now().astimezone()
 
-        executor, invalid_resource_types = await self.setup_executor(model_version, resource_types)
+        executor, invalid_resources = await self.setup_executor(model_version, resource_types)
 
         loaded_resources: list[ResourceDetails] = []
-        failed_resources: list[ResourceVersionIdStr] = []
         undeployable: dict[ResourceVersionIdStr, const.ResourceState] = {}
 
+        logs: dict[str, set[data.LogLine]] = defaultdict(set)
+        failed_resource_ids: dict[str, set[str]] = defaultdict(set)
+        logged_resource_types = set()
```

Comment:
This one seems redundant to me, because it's basically a copy of `logs.keys()`. But if you prefer it style-wise, I'm fine with it.